### DISCUSSION TITLE
Please add a Ratings or a Like feature

### DISCUSSION THREAD

**Discussion by Jason-Kennemer** [copy-pasty/hq#2] [suggestions / feedback]
This looks like a promising app to index code snippets, patterns, ideas, etc. What would make it more fantastic is to add ratings or 'likes' to entries so we can filter or sort based on the rating or count of likes. This should help filter the good from the not-so-great. There could also be a new items tag or section so newly added items with potential receive visibility.

**Comment by ochmanski** [comment]
Hey, thank you for great suggestion!

Context: this project started as more "personal" index but after that kinda viral X comment it makes sense to actually add some sort of review / rating mechanism. There are many approaches we could adopt:

### 1. Detailed review
We could show % of people finding given pattern useful, etc.

Pattern XYZ
70% users found this pattern useful
30% users think it's hard to implement
...

But then if we have small scale the % might look strange? Like 100% users think it's hard to implement, where only one person just gave feedback? I'm sure there's some algorithm making it more balanced at early review stages.

### 2. Simple rating + comments
★ rating or percentage too? Like 80% enjoyed this. Or maybe just likes? 👍 (10) / 👎 (40). Do we want to go YT style where dislikes would not show up in the UI but will affect sorting? Non-bs approach of showing everything could work here better since technical people will going to use it.


Pattern XYZ
★4.5/5
12 comments

User 1: I'm using Next 13 with pages router and it's not really applicable here.
User 2: Switched to this solution from @repo/library, much easier to maintain now!
...

### 3. Hybrid?

We might combine these two so it's easier to provide feedback for people either liking or disliking the resource but also give more vocal people ability to comment and review in a detailed way. Perhaps views count in addition to that might add good insights to potential adopters of a given resource?

Pattern XYZ
Generally useful but hard to implement*
👍 (10) / 👎 (40)
15 comments
2451 views

*60% (140 total) users found this pattern useful, 84% (230 total) users found this pattern hard to implement

### Next steps

We would need to break it up into smaller quick wins (this is a side-project for me) but yeah. Maybe I'm over-engineering with that third approach but it might be useful for less-senior people to quickly see "vibe" of a resource, without having to dive deep into comments or relying solely on like ratio. Having structured insights seems like a great opportunity to showcase them in a cool way some day.
